server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1449,18 +1510,22 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
         try {
             value = value(context.parser(), type, nullValue, coerce());
         } catch (IllegalArgumentException e) {
-            if (ignoreMalformed.value() && context.parser().currentToken().isValue()) {
-                context.addIgnoredField(mappedFieldType.name());
-                return;
-            } else {
-                throw e;
-            }
+            handleIllegalArgumentException(context, e);
```

Comment:
Is this important to the optimization? The exception flow reads less naturally with this extracted.

Reply:
also https://github.com/elastic/elasticsearch/pull/85688#discussion_r857494685 :)